pkg/configuration/filter_resources.go

```diff
@@ -0,0 +1,141 @@
+package configuration
+
+import (
+	"slices"
+
+	"github.com/np-guard/vmware-analyzer/internal/common"
+	"github.com/np-guard/vmware-analyzer/pkg/collector"
+	nsx "github.com/np-guard/vmware-analyzer/pkg/configuration/generated"
+)
+
+func filterResources(rc *collector.ResourcesContainerModel, vms []string) {
+	if len(vms) == 0 {
+		return
+	}
+	rf := resourceFilter{rc: rc, VMs: vms}
+	rf.filterTopology()
+	rf.filterGroups()
+	rf.filterRules()
+}
+
+type resourceFilter struct {
+	rc                  *collector.ResourcesContainerModel
+	VMs                 []string // the filter from the user
+	vmIds               []string // the ids of the filtered the vms
+	vnisAttIds          []string // the vni attach id fo the filtered vnis, use to filter the segment ports
+	vmsAddresses        []string // the vm addresses of the filtered the vms
+	allGroupPaths       []string // all the group paths
+	allRemainGroupPaths []string // the filtered paths
+}
+
+func (f *resourceFilter) filterTopology() {
```

Comment:
missing some tests that demonstrate topology filtering?

Reply:
done

Reply:
what are the relevant tests?

Reply:
Example1dExternalWithSegmentsFiltered
ExampleHogwartsFiltered